### ISSUE TITLE
Release dspy.GRPO optimizer on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [stanfordnlp/dspy#8608]
Hi @TomeHirata 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work on mmGRPO in "Multi-module GRPO: Composing Policy Gradients and Prompt Optimization for Language Model Programs", and your paper was recently featured on Hugging Face Papers: https://huggingface.co/papers/2508.04660. The paper page allows people to discuss your research and find related artifacts. You can also claim the paper as yours, which will be visible on your public profile, and add GitHub and project page URLs.

It's fantastic to see that you've open-sourced `mmGRPO` as the `dspy.GRPO` optimizer within your DSPy framework. While `dspy.GRPO` is an optimizer and not a traditional pre-trained model checkpoint with weights, we are keen to help researchers gain more visibility for their valuable contributions to the AI community.

Would you be interested in exploring options to make the `dspy.GRPO` optimizer (or perhaps specific models optimized with `dspy.GRPO` that you've used for your experiments) more discoverable on the Hugging Face Hub? This could potentially involve creating a "model card" (or a similar community resource card) to showcase the optimizer's capabilities and link directly to your GitHub repository, or if applicable, hosting specific checkpoints of models fine-tuned or optimized using `dspy.GRPO`.

We encourage researchers to share their artifacts on the Hub to improve visibility and discoverability. We can add relevant tags (like "reinforcement-learning" in this case) so that users can easily find this important work.

Let me know if you're interested or need any guidance on how this could be achieved.

Kind regards,

Niels

**Comment by Ziems** [comment]
Hi @NielsRogge! Thanks for reaching out! The way we have things set up with mmGRPO, all of the modules in a DSPy program can be comprised of separate weights and separate prompts. So we can absolutely host the models on huggingface and that is very easy to do. However its not super straightforward to get the prompt stored on huggingface or the program as a whole stored there afaik. This might be a very interesting area to explore. Imagine being able to load a DSPy program (including all the prompts and weights needed) just as easily as you can currently load a HuggingFace model. I'm sure many would find this useful.

**Comment by dilarasoylu** [comment]
Thank you so much @NielsRogge! Seconding @Ziems here, it would be exciting to load DSPy programs right out of HuggingFace!

**Comment by TomeHirata** [comment]
Hi, @NielsRogge. I agree with @Ziems and @dilarasoylu. While it may not be natively supported by huggingface now, it'd be exciting if we can safely host DSPy programs and distribute them.

**Comment by NielsRogge** [comment]
Thanks for your replies! What's the format in which DSPy programs are saved? 

Perhaps DSPY could be added as a native library on the hub: https://huggingface.co/docs/hub/en/models-adding-libraries.

Basically in that case we add `library_name: dspy` to each model/dataset repository's metadata, you can choose custom download metrics and add custom code snippets which appear in the "Use this model" button, like here for a model which is compatible with `library_name: transformers`: https://huggingface.co/openai/gpt-oss-120b?library=transformers.

The custom code snippets live here: https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/model-libraries.ts

**Comment by Ziems** [comment]
Hi @NielsRogge

Thanks for the suggestion! We plan to look into this next week
